registry/spec.md

```diff
@@ -1491,71 +1632,1350 @@ An attempt to delete all versions MUST generate an error.
+[JSON Pointer][JSON Pointer] syntax:
 
-"format": "STRING", ?
-"definitionGroups": [ DEFINITIONGROUP-URI-Reference, ... ] ?
-```
+``` JSONC
+{
+  "$schema": "https://cloudevents.io/schemas/registry",
+  "specversion": "0.4-wip",
+  "id": "urn:uuid:3978344f-8596-4c3a-a978-8fc9a6a469f7",
+  "endpoints" : 
+  {
+    "com.example.telemetry" : {
+      "type": "endpoint",
+      "id" : "com.example.telemetry",
+      "usage": "consumer",
+      "config": {
+        // ... details ......
+      },
+      "format" : "CloudEvents/1.0"
+      "definitiongroups" :[
+        "https://rawdata.repos.example.com/myorg/myproject/main/example.telemetryEvents.cereg#/definitiongroups/com.example.telemetryEvents"
+      ]
+    }
+  }
+}
+```
+
+All other references to other objects in the registry can be expressed in the
+same way.
+
+While the CloudEvents Registry is primarily motivated by enabling development of
+CloudEvents-based event flows, the registry is not limited to CloudEvents. It
+can be used to describe any asynchronous messaging or eventing endpoint and its
+messages, including such that do not use CloudEvents at all. The [Message
+Formats](#message-formats) section therefore not only describes the attribute
+meta-schema for CloudEvents, but also meta-schemas for the native message
+envelopes of MQTT, AMQP, and other messaging protocols.
+
+The registry is designed to be extensible to support any structured data
+encoding and related schemas for message or event payloads. The [Schema
+Formats](#schema-formats) section describes the meta-schema for JSON Schema, XML
+Schema, Apache Avro schema, Protobuf schema.
+
+### File format
+
+A CloudEvents Registry can be implemented using the Registry API or with plain
+text files.
+
+When using the file-based model, files with the extension `.cereg` use JSON
+encoding. Files with the extension `.cereg.yaml` or `.cereg.yml` use YAML
+encoding. The formal JSON schema for the file format is defined in the
+[CloudEvents Registry Document Schema](#cloudevents-registry-document-schema),
+which implements the Registry format and the CloudEvents Registry format.
+
+The media-type for the file format is `application/cloudevents-registry+json` for the
+JSON encoding and `application/cloudevents-registry+yaml` for the YAML encoding.
+
+The JSON schema identifier is `https://cloudevents.io/schemas/registry` and the
+`specversion` property indicates the version of this specification that the
+elements of the file conform to.
+
+A CloudEvents Registry file MUST contain a single JSON object or YAML document.
+The object declares the roots of the three sub-registries, which are either
+embedded or referenced. Any of the three sub-registries MAY be omitted.
+
+``` meta
+{
+   "$schema": "https://cloudevents.io/schemas/registry",
+   "specversion": "0.4-wip",
+   "endpoints": { ... } | "endpointsUrl": "URL",
+   "definitiongroups": { ... } | "definitiongroupsUrl": "URL"
+   "schemagroups" : { ... } | "schemagroupsUrl": "URL",
```

Comment:
Including "schemagroups" here makes sense, but then in the registry metadata we'd need to include its definition in then too, alongside "endpoints" and "definitiongroups". Easy enough, however, does this imply that we should consider some kind of referencing mechanism in the registry metadata section so that people don't need to define/copy some other registry's metadata defs each time they want to include instances of those entities in their registry? Let's discuss

Reply:
remove trailing ,